### ISSUE TITLE
TSL3.1 UDP > TSL5 UDP

### ISSUE THREAD

**Issue by eicdw** [josephdadams/TallyArbiter#220]
Two minor issues, and forgive me if I am mistaken about either.

1. TSL3.1 UDP into Tally Arbiter parses green tallies as red and red as green.
The TSL3.1 source is an Image Video TSI. It shows that it's outputting the correct colors to Arbiter, but in Arbiter,red/green are inverted. I can work around this by inverting the actions I send to the MV (in this case, an EVS XT-VIA's internal multiviewer), so not the biggest issue. This may very well be on the TSI side, but I wanted to mention it since TSL5 from the TSI to Arbiter was not swapped.

![IMG_1554](https://user-images.githubusercontent.com/62229472/134810540-7db61436-38bd-486d-aaf6-659f70bf759e.JPG)

![IMG_1555](https://user-images.githubusercontent.com/62229472/134810544-3fc82936-9cd3-4a0d-bf70-50bb0cd5e67f.JPG)

2. TSL5 UDP should be able to output an amber tally separately from PVW/PGM.
When sending TSL5 UDP commands directly from the TSI, I can send Red, Green, and Amber discreetly in TSL5. It appears I can only send Red and Green from Arbiter. Is it possible to add actions to send amber tally based on incoming TSL3.1?

**Comment by josephdadams** [comment]
What bus does amber represent?

**Comment by josephdadams** [comment]
Also, are you mostly just trying to use Tally Arbiter as a TSL3.1<>TSL5.0 converter?

**Comment by eicdw** [comment]
First, love the software. It's great to see TSL being implemented in a way
that doesn't involve licenses. I'm excited to see how you develop it
further.

Amber tally doesn't necessarily represent a bus...   It's often used for
external tallies that come from sources outside a truck. For example, if we
have two trucks parked next to each other, we will use amber tallies to
show when the other truck tallies a source from our truck. I also allow our
TD to use an aux bus to define which EVS server is "GPI Enabled" - and this
gives that channel an amber tally until they roll/take it.

FWIW, the amber tally should be separate from red/green, rather than the
result of both being active at once as in some devices.

Hope that's helpful, and please let me know if there's any way I can help.

David Williams - EIC 45FLEX
Mobile TV Group
618-655-9937 (c)

**Comment by eicdw** [comment]
Yes. I'm basically trying to tally EVS's built in multiviewers on trucks
that don't have a TSL5 license in Image Video.

David Williams - EIC 45FLEX
Mobile TV Group
618-655-9937 (c)

**Comment by josephdadams** [comment]
This is helpful. I am basing the tsl3.1 PVW/PGM states based on the Ross implementation of the protocol, but we could possibly add a swap feature in the config to swap those bits.

You might be better suited with a direct translator program that just converts between 3.1 and 5. Would be less to manage overall if you’re not needing specifically DIY tally of some kind. Im available for hire to create this if needed.

**Comment by josephdadams** [comment]
I am working on improvements to the TSL Listener Clients section which should resolve this.

<img width="493" height="774" alt="Image" src="https://github.com/user-attachments/assets/bb70770f-3d54-4e75-b2e7-dfe0e1b5a90d" />
<img width="494" height="794" alt="Image" src="https://github.com/user-attachments/assets/eabaf4ff-1d42-48d6-8446-7ff2e0d57607" />
